### PR TITLE
Include Npcap in Packetbeat when bundled with Agentbeat

### PR DESCRIPTION
**Pull request by michalpristas**
## Proposed commit message

Extracting image selector logic into separate package so it can be reused when compiling agent(beat). 
Otherwise npcap installer is not present and packetbeat fails to work when `interfaces!=nil`(trying to install and missing installer)

## Checklist

- [x] My code follows the style guidelines of this project
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have made corresponding change to the default configuration files
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added an entry in `CHANGELOG.next.asciidoc` or `CHANGELOG-developer.next.asciidoc`.

## How to test this PR locally

Package Agentbeat using `NPCAP_LOCAL=true SNAPSHOT=true PLATFORMS=windows/amd64 PACKAGES=zip mage package` 
run `./agentbeat packetbeat run -c packetbeat.yml` on windows (default packetbeat config downloaded from [here](https://github.com/elastic/beats/blob/main/x-pack/packetbeat/packetbeat.yml) is fine) it should complain `any` is not supported on windows. Error `failed to get device list: couldn't load wpcap.dll` should not be present 

This needs to be packaged on Linux system with [msitools](https://launchpad.net/ubuntu/+source/msitools) installed

## Related issues
Closes https://github.com/elastic/elastic-agent/issues/6108

cc @blakerouse

### CODE REVIEW

File: x-pack/packetbeat/scripts/mage/pcap.go [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+	if os.Getenv("CI") != "true" && os.Getenv("NPCAP_LOCAL") != "true" {
+		return image, nil
+	}
+	if platform == "windows/amd64" {
+		image = strings.ReplaceAll(image, "beats-dev", "observability-ci") // Temporarily work around naming of npcap image.
+		image = strings.ReplaceAll(image, "main", "npcap-"+NpcapVersion+"-debian9")
+	}
+	return image, nil
+}
+func CopyNCAPInstaller() error {
```
Comment (andrewkroh) [inline x-pack/packetbeat/scripts/mage/pcap.go:43]: `CopyNCAPInstaller` -> `CopyNPCAPInstaller`

And insert a newline in between the functions please.

**Review by andrewkroh — APPROVED** [review]

**Comment by cmacknz** [comment]
Is it possible to write an integration test to catch this problem?

**Comment by michalpristas** [comment]
what i learned today is that we have integration tests for this. Packetbeat i[mplements it here](https://github.com/elastic/beats/blob/93b018a8362139eeb6b81bde9c43374c7ad1cd46/x-pack/packetbeat/tests/system/app_test.go#L29) and agentbeat is [executing](https://github.com/elastic/beats/blob/main/x-pack/packetbeat/tests/system/app_run_agentbeat_test.go) these tests.
the problem is and also the point where rabbit hole starts running these on windows. as far as i looked no beat is running integration tests on windows, except packetbeat but they are skipped due to [these failures](https://github.com/elastic/beats/issues/38142). this is problem of its own and goes way beyond scope of this PR. i tried to timebox this for today but i'm running into more and more issues preventing to run these. (it's not as simple as adding windows image to buildkite pipeline definition)

cc @ycombinator

**Comment by ycombinator** [comment]
Okay, in that case, I'd be OK with merging this PR as-is provided we prioritize https://github.com/elastic/beats/issues/38142 in an upcoming sprint.  I've asked about that here: https://github.com/elastic/beats/issues/38142#issuecomment-2542003603

**Comment by ycombinator** [comment]
Okay, in that case, I'd be OK with merging this PR as-is provided we prioritize https://github.com/elastic/beats/issues/38142 in an upcoming sprint.  I've asked about that here: https://github.com/elastic/beats/issues/38142#issuecomment-2542003603

@cmacknz @jlind23 WDYT?

**Comment by cmacknz** [comment]
I'm fine with it as long as we have evidence that this works via manual QA, either from our QA team or someone who isn't Michal testing this and posting the results. "Someone who is not the PR author confirmed it worked" has been a pretty strong indicator that things do in fact work. I haven't had a chance to try it myself yet.

**Comment by ycombinator** [comment]
Makes sense, @cmacknz.  Thanks.  

I've added the `QA:Ready for Testing` label and I see @michalpristas has included manual test steps in the PR description. cc: @amolnater-qasource.

**Comment by jlind23** [comment]
@amolnater-qasource were you able to run test on this?

**Review by jlind23 — APPROVED** [review]

**Comment by amolnater-qasource** [comment]
Hi @jlind23 

We have revalidated this PR with latest 8.18.0-SNAPSHOT, 9.0.0-SNAPSHOT artifacts and also created a new package, however we had same results as earlier.

- On running `on running agentbeat.exe packetbeat run -c packetbeat.yml` we observed error: `Exiting: any interface is not supported on windows`.

**Screenshots:**
**9.0 self created package:**
image
image

**8.18-SNAPSHOT:**
image
image

Please let us know if we are missing anything here.

cc: @michalpristas 
Thanks!!

**Comment by amolnater-qasource** [comment]
Hi @ycombinator 

Thank you for sharing the details.
Yes, we are getting only ` Exiting: any interface is not supported on windows` and not any of the other errors.

As the testing is completed on this we are marking this ticket as **QA:Validated**.

Thanks!!

**Pull request merged**